lib/taskana-core-test/src/test/java/acceptance/taskcomment/delete/DeleteTaskCommentAccTest.java

```diff
@@ -0,0 +1,179 @@
+  Workbasket defaultWorkbasket;
+  Task task1;
+  TaskComment comment1;
+
+  @WithAccessId(user = "admin")
+  @BeforeAll
+  void setup() throws Exception {
+    defaultClassification =
+        DefaultTestEntities.defaultTestClassification().buildAndStore(classificationService);
+    defaultWorkbasket =
+        DefaultTestEntities.defaultTestWorkbasket().buildAndStore(workbasketService);
+    WorkbasketAccessItemBuilder.newWorkbasketAccessItem()
+        .workbasketId(defaultWorkbasket.getId())
+        .accessId("user-1-1")
+        .permission(WorkbasketPermission.OPEN)
+        .permission(WorkbasketPermission.READ)
+        .permission(WorkbasketPermission.READTASKS)
+        .permission(WorkbasketPermission.APPEND)
+        .buildAndStore(workbasketService);
+    task1 =
+        TaskBuilder.newTask()
+            .classificationSummary(defaultClassification.asSummary())
+            .workbasketSummary(defaultWorkbasket.asSummary())
+            .primaryObjRef(DefaultTestEntities.defaultTestObjectReference().build())
+            .buildAndStore(taskService);
+  }
+
+  @WithAccessId(user = "user-1-1")
+  @Test
+  void should_DeleteTaskCommentAndUpdateNumberOfComments_For_TaskCommentId() throws Exception {
+    comment1 =
+        TaskCommentBuilder.newTaskComment()
+            .taskId(task1.getId())
+            .textField("Text1")
+            .created(Instant.now())
+            .modified(Instant.now())
+            .buildAndStore(taskService);
+
+    taskService.deleteTaskComment(comment1.getId());
+
+    List<TaskComment> taskCommentsAfterDeletion = taskService.getTaskComments(task1.getId());
+    assertThat(taskCommentsAfterDeletion).hasSize(0);
+    assertThat(taskService.getTask(task1.getId()).getNumberOfComments()).isZero();
+  }
+
+  @WithAccessId(user = "user-1-2", groups = "user-1-1")
+  @Test
+  void should_FailToDeleteTaskComment_When_UserHasNoAuthorization() throws Exception {
+    comment1 =
+        TaskCommentBuilder.newTaskComment()
+            .taskId(task1.getId())
+            .textField("Text1")
+            .created(Instant.now())
+            .modified(Instant.now())
+            .buildAndStore(taskService, "user-1-1");
+    ThrowingCallable call = () -> taskService.deleteTaskComment(comment1.getId());
+    NotAuthorizedOnTaskCommentException e =
+        catchThrowableOfType(call, NotAuthorizedOnTaskCommentException.class);
+    assertThat(e.getTaskCommentId()).isEqualTo(comment1.getId());
+    assertThat(e.getCurrentUserId()).isEqualTo(taskanaEngine.getCurrentUserContext().getUserid());
+
+    List<TaskComment> taskCommentsAfterDeletion = taskService.getTaskComments(task1.getId());
+    assertThat(taskCommentsAfterDeletion).hasSize(1);
+    assertThat(taskService.getTask(task1.getId()).getNumberOfComments()).isOne();
+
+    taskanaEngine.runAsAdmin(
+        () -> {
+          try {
+            taskService.deleteTaskComment(comment1.getId());
+          } catch (TaskCommentNotFoundException ex) {
```

Comment:
di we really need all these catches in a test? there should be no exception at all....

Reply:
okay i will do as you suggested.